containers/src/Kokkos_DualView.hpp

```diff
@@ -875,8 +875,19 @@ class DualView : public ViewTraits<DataType, Arg1Type, Arg2Type, Arg3Type> {
                const size_t n5 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                const size_t n6 = KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                const size_t n7 = KOKKOS_IMPL_CTOR_DEFAULT_ARG) {
-    ::Kokkos::realloc(d_view, n0, n1, n2, n3, n4, n5, n6, n7);
-    h_view = create_mirror_view(d_view);
+    const size_t new_extents[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
+    bool sizeMismatch           = false;
+    for (unsigned int dim = 0; dim < h_view.rank_dynamic; ++dim)
+      if (new_extents[dim] != h_view.extent(dim)) {
+        sizeMismatch = true;
+        break;
+      }
+
+    if (sizeMismatch) {
+      ::Kokkos::realloc(d_view, n0, n1, n2, n3, n4, n5, n6, n7);
+      h_view = create_mirror_view(d_view);
+    } else
+      ::Kokkos::deep_copy(d_view, typename t_dev::value_type{});
```

Comment:
Why did you add this else-branch?
It was not present in the original code.

Reply:
Same here. The previous behavior was that `realloc` zero initializes the View. I am, of course, open to discuss not doing it. So far my thinking was to avoid omit it only when we pass `WithoutInitializing` (which is a later pull request someone else was assigned to IIRC).

Reply:
Got it

Reply:
ok